src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -336,6 +337,60 @@ public static List<string> LookupNames(this SemanticModel model, int position, I
             return symbols.Select(s => s.Name).Distinct().ToList();
         }
 
+        internal static TypeInfo GetTypeInfoAndVerifyIOperation(this SemanticModel model, SyntaxNode expression)
+        {
+            var typeInfo = model.GetTypeInfo(expression);
+            var iop = getOperation(model, expression);
+            if (typeInfo.Type is null)
+            {
+                Assert.True(iop?.Type is null ||
+                            iop is ITupleOperation { NaturalType: null });
+            }
+            else if (iop is { Type: { } })
+            {
+                Assert.Equal(typeInfo.Type.NullableAnnotation, iop.Type.NullableAnnotation);
+            }
+            else
+            {
+                Assert.True(isValidDeclaration(expression));
+
+                static bool isValidDeclaration(SyntaxNode expression)
+                    => (expression.Parent is VariableDeclarationSyntax decl && decl.Type == expression) ||
+                       (expression.Parent is ForEachStatementSyntax forEach && forEach.Type == expression) ||
+                       (expression.Parent is DeclarationExpressionSyntax declExpr && declExpr.Type == expression) ||
+                       (expression.Parent is RefTypeSyntax refType && isValidDeclaration(refType));
+            }
+
+            if (iop is { Parent: IConversionOperation parentConversion })
+            {
+                iop = parentConversion;
+            }
+
+            if (typeInfo.ConvertedType is null)
+            {
+                Assert.Null(iop?.Type);
+            }
+            else if (iop is { Type: { } })
+            {
+                Assert.Equal(typeInfo.ConvertedType.NullableAnnotation, iop.Type.NullableAnnotation);
+            }
+
+            return typeInfo;
+
+            static IOperation getOperation(SemanticModel model, SyntaxNode expression)
+            {
+                // Nullable suppressions are not directly represented in the bound tree. Rather, they are set
+                // as flags on the bound node underlying the node. Therefore, there is similarly no representation
+                // in the IOperation tree, and we should retrieve the IOperation node underlying the suppression.
+                if (expression.IsKind(SyntaxKind.SuppressNullableWarningExpression))
+                {
+                    expression = ((PostfixUnaryExpressionSyntax)expression).Operand;
```

Comment:
>expression = ((PostfixUnaryExpressionSyntax)expression).Operand; [](start = 20, length = 64)

Should we do this in a loop and, perhaps, dig through parenthesis too?

Reply:
I believe this was originally in `getOperation`, I am proceeding with that assumption.

---
In reply to: [402673404](https://github.com/dotnet/roslyn/pull/42403#discussion_r402673404) [](ancestors = 402673404)